pkg/hooks/hooks_test.go

```diff
@@ -77,3 +79,62 @@ func testManifest() pods.Manifest {
 	builder.SetID(podId)
 	return builder.GetManifest()
 }
+
+func TestHookRunWithTimeout(t *testing.T) {
+	type sleepTest struct {
+		sleep      int
+		timeout    int
+		shouldPass bool
+	}
+
+	var sleepTests = []sleepTest{
+		{1, 2, true},
+		{2, 1, false},
```

Comment:
If you use a `time.Duration` you could make this test run in acceptable time with `100ms` vs `50ms` or similar. Mike's suggestion is strictly less flimsy than mine though :)